File: src/main/java/jv/com/tencent/cain/design_pattern/simple_factory/Operation.java
package jv.com.tencent.cain.design_pattern.simple_factory;

/**
 * @author cainjiang
 * @date 2019-02-21
 * 操作的基类
 */
abstract public class Operation {
    /**
     * 操作数a
     */
    public double numA;
    /**
     * 操作数b
     */
    public double numB;

    public double getNumA() {
        return numA;
    }

    public double getNumB() {
        return numB;
    }

    /**
     * 获取操作结果
     *
     * @return
     */
    protected abstract double getResult();
}

File: src/main/kotlin/kt/com/tencent/cain/desigin_pattern/simple_factoty/Operation.kt
package kt.com.tencent.cain.desigin_pattern.simple_factoty

/**
 * @author cainjiang
 * @date 2019-02-22
 * 操作的基类
 */
abstract class Operation {
    var numA: Double = 0.0
    var numB: Double = 0.0
    abstract fun getResult(): Double
}